src/shared_utilities/drivers.jl

```diff
@@ -400,26 +467,55 @@ end
                         T_sfc,
                         )
 
-A helper function which returns the surface air density for a given
-model, needed because different models compute and store surface air density
- in different ways and places.
+A helper function which returns the surface air density; this assumes that
+the `model` has a property called `parameters` containing `earth_param_set`.
 
 We additionally include the `atmos` type as an argument because
-the surface air density computation may change between a coupled simulation
+the surface air density computation will change between a coupled simulation
 and a prescibed atmos simulation.
 
 Extending this function for your model is only necessary if you need to
-compute surface fluxes and radiative fluxes at the surface using
-the functions in this file.
+compute the air density in a different way.
 """
 function surface_air_density(
-    atmos::AbstractAtmosphericDrivers,
+    atmos::PrescribedAtmosphere,
     model::AbstractModel,
     Y,
     p,
     t,
     T_sfc,
-) end
+)
+    thermo_params =
+        LSMP.thermodynamic_parameters(model.parameters.earth_param_set)
+    ts_in = construct_atmos_ts(atmos, p, thermo_params)
+    return compute_ρ_sfc.(thermo_params, ts_in, T_sfc)
+end
+
+
+"""
+    ClimaLSM.surface_air_density(
+                    atmos::CoupledGlobalAtmosphere,
+                    model::AbstractModel,
+                    Y,
+                    p,
+                    _...,
+                )
+Returns the air density at the surface in the case of a coupled simulation.
+
+This requires the field `ρ_sfc` to be present in the cache `p` under the name
```

Comment:
ideally we should be able to handle this with dispatch when setting up individual model cache variables.